docs/tasks.md

```diff
@@ -992,112 +992,45 @@ The following example `Task` builds and pushes a `Dockerfile`-built image.
 unsafe** and is shown here only as a demonstration. Use [kaniko](https://github.com/GoogleContainerTools/kaniko) instead.
 
 ```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
 spec:
   params:
-    # These may be overridden, but provide sensible defaults.
-    - name: directory
-      type: string
-      description: The directory containing the build context.
-      default: /workspace
+    # This may be overridden, but is a sensible default.
     - name: dockerfileName
       type: string
       description: The name of the Dockerfile
       default: Dockerfile
-  resources:
-    inputs:
-      - name: workspace
-        type: git
-    outputs:
-      - name: builtImage
-        type: image
+    - name: image
+      type: string
+      description: The image to build and push
+  workspaces:
+  - name: source
+  - name: docker-socket
+    mountPath: /var/run/docker.sock
   steps:
     - name: dockerfile-build
       image: gcr.io/cloud-builders/docker
-      workingDir: "$(params.directory)"
+      workingDir: "$(workspaces.source.path)"
       args:
         [
           "build",
           "--no-cache",
           "--tag",
-          "$(resources.outputs.image.url)",
+          "$(params.image)",
           "--file",
           "$(params.dockerfileName)",
           ".",
         ]
-      volumeMounts:
-        - name: docker-socket
-          mountPath: /var/run/docker.sock
-
     - name: dockerfile-push
       image: gcr.io/cloud-builders/docker
-      args: ["push", "$(resources.outputs.image.url)"]
-      volumeMounts:
-        - name: docker-socket
-          mountPath: /var/run/docker.sock
-
-  # As an implementation detail, this Task mounts the host's daemon socket.
-  volumes:
-    - name: docker-socket
-      hostPath:
-        path: /var/run/docker.sock
-        type: Socket
-```
-
-#### Mounting multiple `Volumes`
```

Comment:
As long as the feature is still there perhaps we shouldn't fully remove this docs, but alter them to say that while this is possible, it is not recommended unless in use cases where no alternative exists.